### ISSUE TITLE
Replace entity manager handler with doctrine resettable em bundle

### ISSUE THREAD

**Issue by Rastusik** [k911/swoole-bundle#185]
**Is your feature request related to a problem? Please describe.**
The problem of the implemented entity manager handler is, that when an exception occurs in the entity manager, it gets closed and it won't be reset automatically. 

**Describe the solution you'd like**
I've implemented a resettable entity manager handler in this bundle: https://github.com/pixelfederation/doctrine-resettable-em-bundle . I would like to remove the current entity manager handler and suggest to integrate the bundle. 

**Describe alternatives you've considered**
Normally I would just made a pull request to get the bundle's code into this swoole bundle, but the problem is, that I needed the resetting outside of swoole bundle (in Symfony Messenger worker context) . 

What do you think, @k911  ?

In addition to that (not sure whether I should open another issue, so I'm briefly mentioning it here) - I have a fork of your repository (https://github.com/pixelfederation/swoole-bundle) and I would like to port the features from the fork into your bundle. Feature after feature, separately, so the process and the features are understandable. What do you think about that? I couldn't do it before, because I needed to make some experiments and get our app up and running, so it was faster to proceed separately.

**Comment by k911** [comment]
I don't think there is a need to port anything right now, because that would only complicate things. TBH, I'd like to get rid of current support for Doctrine Entity Manager too. I'm currently working on a more flexible configuration with the ability to dynamically attach handlers to different server events, so anyone could easily attach their own features by just implementing the interface.

Example configuration will look like this:

```yaml
services:
  MyCustomRequestHandler:
    class: Vendor/RequestHandlers/MyCustomRequestHandler

swoole:
  server:
    listeners:
      - id: http 
        host: localhost
        protocol: tcp_dualstack
        port: 9501
        handlers:
          request:
           # predefined request handlers
           - symfonyRequestHandler
           - id: 'MyCustomRequestHandler' # reference to service in Symfony DI container
             priority: 200
          workerStart:
          - #... 
      # predefined listener
      - id: api
      # predefined listener with some changes
      - id: websocket
        tls:
           enabled: true
           key: 'key.pem'
           cert: 'cert.pem'
           ca: 'ca.pem'
          

```

Class `Vendor/RequestHandlers/MyCustomRequestHandler` would need to be even in user-space or in another Symfony bundle. It will just need to implement `RequestHandlerInterface`.

I'm working on these features on branch `feat/dynamic-listeners` if you'd like to try.

To summarize all of that:

* There is no need to port anything regarding Doctrine project inside this bundle
* There will be an easier configuration of custom event handlers for Swoole server's events (like handling request, tasks, etc.) so anyone could add the things they like
* What I'm really interested in the future is the documentation how to easily integrate external bundles, like yours @Rastusik

**Comment by Rastusik** [comment]
I think I didn't explain clearly what I meant. My point was to remove doctrine code from this bundle and create something like a bridge, or what you call a custom request handler to call the doctrine code from this bundle: https://github.com/pixelfederation/doctrine-resettable-em-bundle . I think it is quite similar to your suggestion and maybe we should work out the details. Or do you see a bigger difference?

**Comment by k911** [comment]
Yeah, I can agree to that, apart from removing things. I'd like to keep backward compatibility for some time, then deprecate and eventually remove it. Because it won't block us to introduce a new feature.

**Comment by Rastusik** [comment]
Ok, the removal can be postponed, no problem. So are you ok with me making a pull request for the mentioned integration? If yes, should I work with  the `feat/dynamic-listeners` branch then?

**Comment by k911** [comment]
Nope, as usual you should work with `develop`.
